src/framework/uicomponents/view/validators/intinputvalidator.cpp

```diff
@@ -48,8 +48,10 @@ QValidator::State IntInputValidator::validate(QString& inputStr, int& cursorPos)
 {
     QValidator::State state = Invalid;
 
-    if (inputStr.contains(QRegularExpression("^\\-?\\d{1,3}$"))) {
-        if (inputStr.contains(QRegularExpression("^\\-?0{2,3}"))
+    //maxDigits is the number of digits in the maximum absolute value between m_top and m_bottom.
+    int maxDigits = std::ceil(std::log10(std::max(std::abs(m_top), std::abs(m_bottom))));
+    if (inputStr.contains(QRegularExpression(QString("^\\-?\\d{1,%1}$").arg(maxDigits)))) {
+        if (inputStr.contains(QRegularExpression(QString("^\\-?0{2,%1}").arg(maxDigits)))
```

Comment:
Wait, I realised that this might cause problems if maxDigits is 1. I guess that the expression will be invalid in that case. Easiest fix seems:
```suggestion
        if (maxDigits >= 2 && inputStr.contains(QRegularExpression(QString("^\\-?0{2,%1}").arg(maxDigits)))
```

Reply:
Thank you for noticing. I'll update it.

Reply:
{2,2} seems to be valid according to the information I found online.